speech_recognition/__init__.py

```diff
@@ -716,6 +716,91 @@ def recognize_google(self, audio_data, key = None, language = "en-US", show_all
+
+        Returns the most likely transcription if ``show_all`` is False (the default). Otherwise, returns the raw API response as a JSON dictionary.
+
+        Raises a ``speech_recognition.UnknownValueError`` exception if the speech is unintelligible. Raises a ``speech_recognition.RequestError`` exception if the speech recognition operation failed, if the credentials aren't valid, or if there is no Internet connection.
+        """
+        if speech_context is None:
+            speech_context = []
+
+        assert isinstance(audio_data, AudioData), "`audio_data` must be audio data"
+        assert isinstance(language, str), "`language` must be a string"
+        assert isinstance(filter_profanity, bool), "`filter_profanity` must be a bool"
+        assert isinstance(speech_context, list), "`speech_context` must be a list"
+
+        # See https://cloud.google.com/speech/reference/rest/v1beta1/RecognitionConfig
+        flac_data = audio_data.get_flac_data(
+            # Audio samples must be at least 8 kHz and at most 48 kHz. Do not
+            # convert if in the range; if outside of it convert to clamped to
+            # that range.
+            convert_rate=None if 8000 <= audio_data.sample_rate <= 48000
+            else max(8000, min(audio_data.sample_rate, 48000)),
+            convert_width=2  # audio samples must be 16-bit
+        )
+
+        speech_service = self.get_speech_service()
+        request = speech_service.speech().syncrecognize(body={
+            "audio": {
+                "content": base64.b64encode(flac_data).decode("utf8"),
+            },
+            "config": {
+                "encoding": "FLAC",
+                "sampleRate": audio_data.sample_rate,
+                "languageCode": language,
+                "profanityFilter": filter_profanity,
+                "speechContext": {
+                    "phrases": speech_context,
+                },
+            },
+        })
+
+        import googleapiclient.errors
+        try:
+            response = request.execute()
+        except googleapiclient.errors.HttpError as e:
+            raise RequestError(e)
+        except URLError as e:
+            raise RequestError("recognition connection failed: {0}".format(e.reason))
+
+        if show_all:
+            return response
+
+        if "results" not in response or len(response["results"]) == 0:
+            raise UnknownValueError()
+
+        transcript = ""
+        for result in response["results"]:
+            transcript += result['alternatives'][0]["transcript"].strip() + " "
+
+        return transcript
+
+    @staticmethod
+    def get_speech_service():
+        try:
+            from oauth2client.client import GoogleCredentials
+            from googleapiclient.discovery import build
+
+            credentials = GoogleCredentials.get_application_default()
+
+            return build("speech", "v1beta1", credentials=credentials)
+        except ImportError:
```

Comment:
Would be nice to capture the message as `except ImportError as error` and then print it to the output. Today I got this error, and it was not just missing the `google-api-python-client`.

Reply:
Wow, yeah, why did I do that? This PR has already been merged for years; I'd certainly advocate for a PR to remove the try-except here. If you open one and want something from me, please let me know!